options-design/mongo/mongo.go

```diff
@@ -0,0 +1,124 @@
+// method on the Database type can be used to execute arbitrary commands
+// against a database.
+type Client struct{}
+
+// NewClient creates a client from the given uri string.
+func NewClient(uri string, opts ...clientopts.Option) (*Client, error) { return nil, nil }
+
+// Database returns a handle to a MongoDB database.
+func (c *Client) Database(name string) *Database { return nil }
+
+// Database is a handle to a specific database within MongoDB. It can be used to retrieve
+// handles to collections within the database and to perform arbitrary commands
+// on the database via the RunCommand method.
+type Database struct{}
+
+// Name returns the name of this database.
+func (db *Database) Name() string { return "" }
+
+// Collection creates a handle to the given collection within this database.
+func (db *Database) Collection(name string) *Collection { return nil }
+
+// RunCommand allows the execution of arbitrary commands against the database.
+// RunCommand always returns a non-nil value. Errors are deferred until
+// RunCommandResult's Decode method is called. This allows easy chaining of the
+// methods.
+func (db *Database) RunCommand(ctx context.Context, cmd interface{}) *DocumentResult { return nil }
+
+// Collection is a handle to a specific collection within a MongoDB database.
+type Collection struct{}
+
+// InsertOne inserts the provided document. If the provided document does not
+// have an "_id" field, one will be generated.
+func (c *Collection) InsertOne(ctx context.Context, doc interface{}, opts ...option.InsertOneOptioner) (*InsertOneResult, error) {
+	return nil, nil
+}
+
+// InsertMany inserts the provided documents. If any of the documents are
+// missing an "_id" field, one will be generated.
+func (c *Collection) InsertMany(ctx context.Context, docs []interface{}, opts ...option.InsertManyOptioner) (*InsertManyResult, error) {
+	return nil, nil
+}
+
+// DeleteOne deletes a single document.
+func (c *Collection) DeleteOne(ctx context.Context, filter interface{}, opts ...option.DeleteOptioner) (*DeleteResult, error) {
+	return nil, nil
+}
+
+// DeleteMany deletes multiple documents.
+func (c *Collection) DeleteMany(ctx context.Context, filter interface{}, opts ...option.DeleteOptioner) (*DeleteResult, error) {
+	return nil, nil
+}
+
+// UpdateOne updates a single document.
+func (c *Collection) UpdateOne(ctx context.Context, filter interface{}, update interface{}, opts ...option.UpdateOptioner) (*UpdateResult, error) {
+	return nil, nil
+}
+
+// UpdateMany updates multiple documents.
+func (c *Collection) UpdateMany(ctx context.Context, filter interface{}, update interface{}, opts ...option.UpdateOptioner) (*UpdateResult, error) {
+	return nil, nil
+}
+
+// ReplaceOne replaces a single document.
+func (c *Collection) ReplaceOne(ctx context.Context, filter interface{}, replacement interface{}, opts ...option.UpdateOptioner) (*UpdateResult, error) {
```

Comment:
I would suggest using ReplaceOptioner. I seem to recall something that is unique about Replace, or was being considered. Might as well make a break to ensure no problems down the road.

Reply:
Yes, there's a drivers ticket about this: https://jira.mongodb.org/browse/DRIVERS-416

Reply:
This is actually implemented properly in the driver code, this is a typo.